src/AiPlayer.js

```diff
@@ -56,12 +56,60 @@ var ticTacToe = ticTacToe || {};
     }
   };
 
-  app.AiPlayer.prototype.calculateMove = function (board, side) {
+  app.AiPlayer.prototype.calculateMove = function (board, me) {
+    var enemy = this.getEnemy(me);
+    // stub to make the Ai always return a move...
+    var square = this.isSquareAvailable(board);
+    if (square !== false) return square;
+    // end stub. remove when coding commences.
+    this.isSquareAvailable(board);
+    var twoInARow = this.findTwoInARowWithEmptyOption(board, me);
+    if (twoInARow !== false) return twoInARow;
+    return false;
+    throw new Error('could not decide on a move');
+  };
+
+  app.AiPlayer.prototype.isSquareAvailable = function (board) {
     for (var i = 0; i < this.getNumberOfSquares(); i++) {
       if (board[i] === 0) {
-        return i;
+        return board[i];
       }
     }
-  };
+    throw new Error("there are no available squares to play");
+  }
+
+  app.AiPlayer.prototype.findTwoInARowWithEmptyOption = function (board, side) {
+    var vertically = this.findTwoInARowVerticallyWithEmptyOption(board, side);
+    if (vertically !== false) return vertically;
+
+    return 1;
+  }
+
+  app.AiPlayer.prototype.findTwoInARowVerticallyWithEmptyOption = function (board, side) {
+    var verticals = [];
+
+    verticals[0] = [board[0], board[3], board[6]];
+    verticals[1] = [board[1], board[4], board[7]];
+    verticals[2] = [board[2], board[5], board[8]];
+    var count = verticals.length;
```

Comment:
remember JavaScript has "hoisting". If we console.log(count) above this line it will actually return "undefined", because the variable is hoisted to the top of the scope (which in this case is the findTwoInARowVerticallyWithEmptyOption method). I suggest just declaring var count; at the top then setting count = verticals.length; on this line